pkg/k8s/ipresolver.go

```diff
@@ -161,23 +193,68 @@ func (resolver *K8sIPResolver) StartWatching() error {
 				deploymentsWatcher.Stop()
 				cronJobsWatcher.Stop()
 				return
-			case podEvent := <-podsWatcher.ResultChan():
+			case podEvent, ok := <-podsWatcher.ResultChan():
+				podsCounter.Inc()
+				if !ok {
+					podsWatcher, err = resolver.clientset.CoreV1().Pods("").Watch(context.Background(), metav1.ListOptions{})
```

Comment:
Need to check for errors here. Should be the same behavior as the first initialization - quit with an error

Reply:
I'm not sure I agree. In case of error I want to keep trying, maybe the API server is down for a moment

Reply:
Then maybe some sleep-until-retry is needed - this method will spam the API server

Reply:
done